### PR TITLE
K8SPXC-1005 document storing backups on azure

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: docs/backups.md [hunk 1]
```diff
@@ -363,7 +445,23 @@ restoration can be done in the following way.
...
                 region: us-west-2
                 endpointUrl: https://URL-OF-THE-S3-COMPATIBLE-STORAGE
+                ...
+            ```
+
+        === "Azure Blob storage"
+
+            The `backupSource` key should contain `destination` key equal to the
+            Azure Blob [container](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers)
+            with a special `azure://` prefix, followed by the necessary Azure
```
Comment (fiowro) [inline docs/backups.md:455]:
```suggestion
            with `https://` prefix, followed by the necessary Azure
```

File: docs/backups.md [hunk 2]
```diff
@@ -363,7 +445,23 @@ restoration can be done in the following way.
...
+            configuration keys, same as in `deploy/cr.yaml` file:
+
+            ```yaml
             ...
+            backupSource:
+              destination: azure://AZURE-CONTAINER-NAME/BACKUP-NAME
```
Comment (fiowro) [inline docs/backups.md:461]:
```suggestion
              destination: https://AZURE-CONTAINER-NAME/BACKUP-NAME
```

File: docs/backups.md [hunk 3]
```diff
@@ -363,7 +445,23 @@ restoration can be done in the following way.
...
+
+            The `backupSource` key should contain `destination` key equal to the
+            Azure Blob [container](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers)
+            with `https://` prefix, followed by the necessary Azure
+            configuration keys, same as in `deploy/cr.yaml` file:
+
+            ```yaml
             ...
+            backupSource:
+              destination: https://AZURE-CONTAINER-NAME/BACKUP-NAME
```
Comment (egegunes) [inline docs/backups.md:461]: @nmarukovich please verify this

**Review by egegunes — APPROVED** [review]
LGTM, but I'm not sure about destination value in restore with backupSource on Azure.

File: docs/backups.md (hunk 3 above)
Comment (nmarukovich) [reply docs/backups.md:461]: The whole way looks like this 
https://engk8soperators.blob.core.windows.net/operator-testing/cluster1-2022-11-21-17:51:07-full

operator-testing = AZURE-CONTAINER-NAME

Comment (pooknull) [reply docs/backups.md:461]: This should not work with `https://` prefix

Comment (egegunes) [reply docs/backups.md:461]: @pooknull is `https://` not necessary or just wrong?

Comment (nmarukovich) [reply docs/backups.md:461]:
```
apiVersion: pxc.percona.com/v1
kind: PerconaXtraDBClusterRestore
metadata:
  name: restore1
spec:
  pxcCluster: cluster1
  backupSource:
    destination: operator-testing/cluster1-2022-12-06-19:39:19-full
    azure:
      credentialsSecret: azure-secret
      container: operator-testing
      endpointUrl: https://engk8soperators.blob.core.windows.net
```

Comment (pooknull) [reply docs/backups.md:461]: @egegunes It's wrong. We shouldn't use any prefixes for that

Comment (fiowro) [inline docs/backups.md:461]:
```suggestion
              destination: AZURE-CONTAINER-NAME/BACKUP-NAME
```

File: docs/backups.md [hunk 4]
```diff
@@ -363,7 +445,23 @@ restoration can be done in the following way.
...
                 region: us-west-2
                 endpointUrl: https://URL-OF-THE-S3-COMPATIBLE-STORAGE
+                ...
+            ```
+
+        === "Azure Blob storage"
+
+            The `backupSource` key should contain `destination` key equal to the
+            Azure Blob [container](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers)
+            with `https://` prefix, followed by the necessary Azure
```
Comment (fiowro) [inline docs/backups.md:455]:
```suggestion
            name, followed by the necessary Azure
```

Comment (fiowro) [inline docs/backups.md:455]:
```suggestion
            name, followed by the backup name
```

File: docs/backups.md [hunk 5]
```diff
@@ -363,7 +445,23 @@ restoration can be done in the following way.
...
+            configuration keys, same as in `deploy/cr.yaml` file:
```
Comment (fiowro) [inline docs/backups.md:456]:
```suggestion
            
```

File: docs/backups.md (hunk 4 above)
Comment (fiowro) [inline docs/backups.md:455]:
```suggestion
            and backup name, followed by the necessary Azure
```

**Review by nmarukovich — APPROVED** [review]

**Review by pooknull — APPROVED** [review]

**Pull request merged**